pkg/roachprod/opentelemetry/files/opentelemetry-config.yaml.tmpl

```diff
@@ -0,0 +1,122 @@
+---
+extensions:
+  health_check:
+
+receivers:
+  {{- if .CockroachDBMetrics }}
+  prometheus/cockroachdb:
+    config:
+      global:
+        scrape_interval: 30s
+      scrape_configs:
+      - job_name: cockroachdb
+        honor_timestamps: false
+        metrics_path: /_status/vars
+        tls_config:
+          insecure_skip_verify: true
+        follow_redirects: true
+        enable_http2: true
+        static_configs:
+          # Multiple targets are listed to cover the cases when the default
+          # HTTP port is used or when scanning for a free HTTP port starting
+          # from 29000. In the latter case 29000 will be used for SQL and 29001
+          # for HTTP.
+          - targets:
+            - "localhost:26258"
+            - "localhost:29001"
+  {{- end }}
+  hostmetrics:
+    collection_interval: 10s
+    scrapers:
+      paging:
+        metrics:
+          system.paging.utilization:
+            enabled: true
+      cpu:
+        metrics:
+          system.cpu.utilization:
+            enabled: true
+      disk:
+      filesystem:
+        metrics:
+          system.filesystem.utilization:
+            enabled: true
+      load:
+      memory:
+      network:
+      processes:
+
+processors:
+  batch/datadog:
+    send_batch_max_size: 1000
+    send_batch_size: 100
+    timeout: 10s
+  memory_limiter/datadog:
+    check_interval: 1s
+    limit_mib: 1024
+  {{- if .CockroachDBMetrics }}
+  filter/datadog:
+    metrics:
+      include:
+        match_type: strict
+        metric_names:
+        {{- range $key, $value := .CockroachDBMetrics }}
+        - {{ $key }}
+        {{- end }}
+  metricstransform/datadog:
+    transforms:
+    {{- range $key, $value := .CockroachDBMetrics }}
+    - include: {{ $key }}
+      match_type: strict
+      action: update
+      new_name: cockroachdb.{{ $value }}
+    {{- end }}
+  {{- end }}
+  {{- if .DatadogTags }}
+  attributes/datadog:
+    actions:
+    {{- range $key, $value := .DatadogTags }}
+    - key: "{{ $key }}"
+      value: "{{ $value }}"
+      action: upsert
+    {{- end }}
+  {{- end }}
+
+exporters:
+  datadog:
+    api:
+      key: ${env:DD_API_KEY}
+      site: ${env:DD_SITE}
+
+service:
+  pipelines:
+    metrics:
+      receivers:
+      {{- if .CockroachDBMetrics }}
```

Comment:
I wonder if we can drop the conditional and always send crdb metrics if running otelcol. But if we want to keep the conditional, maybe wrap this entire pipeline instead of the receiver? Would this even be a valid pipeline with zero receivers?

Reply:
I'm going to wrap the pipeline. I added these conditionals since I don't want to have a case where we're sending CockroachDB metrics without the rename and explode a Datadog bill.